crates/build/re_types_builder/src/codegen/fbs/mod.rs

```diff
@@ -0,0 +1,90 @@
+//! Codegen fbs file that are used for subsequent codegen steps.
+
+use camino::{Utf8Path, Utf8PathBuf};
+
+use crate::{CodeGenerator, GeneratedFiles};
+
+use super::autogen_warning;
+
+pub struct FbsCodeGenerator {
+    definition_dir: Utf8PathBuf,
+}
+
+impl FbsCodeGenerator {
+    pub fn new(definition_dir: impl Into<Utf8PathBuf>) -> Self {
+        Self {
+            definition_dir: definition_dir.into(),
+        }
+    }
+
+    fn add_include_for(
+        &mut self,
+        reporter: &crate::Reporter,
+        files_to_write: &mut std::collections::BTreeMap<Utf8PathBuf, String>,
+        directory: &str,
+    ) {
+        files_to_write.insert(
+            self.definition_dir.join(format!("{directory}.fbs")),
+            generate_include_file_for_dir(reporter, &self.definition_dir.join(directory)),
+        );
+    }
+}
+
+impl CodeGenerator for FbsCodeGenerator {
+    fn generate(
+        &mut self,
+        reporter: &crate::Reporter,
+        _objects: &crate::Objects,              // Expected to be empty.
+        _arrow_registry: &crate::ArrowRegistry, // Expected to be empty.
+    ) -> crate::GeneratedFiles {
+        let mut files_to_write = GeneratedFiles::default();
+
+        self.add_include_for(reporter, &mut files_to_write, "attributes");
+
+        self.add_include_for(reporter, &mut files_to_write, "rerun/datatypes");
+        self.add_include_for(reporter, &mut files_to_write, "rerun/components");
+        self.add_include_for(reporter, &mut files_to_write, "rerun/archetypes");
+
+        self.add_include_for(reporter, &mut files_to_write, "rerun/blueprint/datatypes");
+        self.add_include_for(reporter, &mut files_to_write, "rerun/blueprint/components");
+        self.add_include_for(reporter, &mut files_to_write, "rerun/blueprint/archetypes");
+        self.add_include_for(reporter, &mut files_to_write, "rerun/blueprint/views");
+
+        files_to_write
+    }
+}
+
+fn generate_include_file_for_dir(reporter: &crate::Reporter, dir_path: &Utf8Path) -> String {
+    let mut contents = format!("// {}", autogen_warning!());
+    contents.push_str("\n\n");
+
+    let read_dir = match dir_path.read_dir() {
+        Ok(read_dir) => read_dir,
+        Err(err) => {
+            reporter.error_file(dir_path, err);
+            return contents;
+        }
+    };
+
+    let dir_name = dir_path.file_name().unwrap();
+
+    for entry in read_dir {
```

Comment:
The order you get here might be OS dependent? Could that be the cause of the error?

Reply:
👍 let's enforce alphabetical ordering

Reply:
that was it, nice catch!